### DISCUSSION TITLE
Building with less librarries, aka without zlib

### DISCUSSION THREAD

**Discussion by aosenkidu** [zuri-lang/zuri#144] [Q&A] [answered]
Is there a way to reduce the required libs?
CMake gives the following error: 
   Could NOT find ZLIB (missing: ZLIB_LIBRARY) (found version "1.2.11")

So, zlib is there, but seems not to be the right version?
I installed a newer version with homebrew on my mac, same error.
Any idea how to circumvent this?

I'm primarily interested in fooling with the language and the vm, not with all the ecosystem around it.

The whole project looks cool  :D 

Best Regards

**Comment by mcfriend99** [comment]
Looks more like a CMake problem. Which CMake version are you using??

**Reply by aosenkidu** [reply]
My cmake is 3.25.1. 
I'm on an relatively old macOS, 10.4.1 Mojave.

**Comment by mcfriend99** [comment]
I can't reproduce this issue despite all my attempts. Can you try with a lower version of cmake??
Let me know if you were able to resolve this.

**Reply by benstigsen** [reply, marked as answer]
I had the same issues when cloning and trying to build on Windows 10, both with `zlib` but also `openssl`.
If you just want to build Blade, without the packages, you can edit `packages/CMakeLists.txt` and remove:
```
add_subdirectory(json)
add_subdirectory(sqlite)
add_subdirectory(ssl)
add_subdirectory(curl)
add_subdirectory(zlib)
if(NOT DISABLE_CLIB)
  add_subdirectory(clib)
endif()
```

**Reply by mcfriend99** [reply]
I'll work on making both `zlib` and `ssl` module configurable during CMake build just like `clib` currently is.
